concourse/steps/scan_sources.py

```diff
@@ -197,54 +199,99 @@ def scan_component_with_whitesource(whitesource_cfg_name: str,
-        ci.util.info('preparing POST project for {}...'.format(component.name()))
+        ci.util.info('preparing POST project for {}'.format(component.name()))
         post_project_object = get_post_project_object(
             whitesource_client=client,
             product_token=product_token,
             component=component)
-        ci.util.info('POST project prepared')
 
         # store in tmp file
         with tempfile.TemporaryFile() as tmp_file:
 
-            ci.util.info('guessing commit hash...')
+            ci.util.info('guessing commit hash')
             # guess git-ref for the given component's version
             commit_hash = product.util.guess_commit_from_ref(component)
-            ci.util.info('commit hash guessed')
 
             # download whitesource component
-            ci.util.info('downloading component for scan...')
+            ci.util.info('downloading component for scan')
             whitesource.component.download_component(
                 github_api=post_project_object.github_api,
                 component_name=post_project_object.component_name,
                 dest=tmp_file,
                 ref=commit_hash)
-            ci.util.info('component downloaded')
 
             # POST component>
-            ci.util.info('POST project...')
+            ci.util.info('POST project')
             post_project_object.whitesource_client.post_product(
                 product_token=post_project_object.product_token,
                 component_name=post_project_object.component_name.name(),
+                component_version=post_project_object.component_version,
                 requester_email=requester_mail,
                 extra_whitesource_config=extra_whitesource_config,
                 file=tmp_file)
-            ci.util.info('project posted\n')
+
+        # get all projects
+        ci.util.info('retrieving all projects')
+        projects = client.get_all_projects(product_token=product_token)
+
+        report = whitesource.util.find_greatest_cve(client=client,
+                                                    projects=projects)
+
+        # generate reporting table for console
+        tables = whitesource.util.generate_reporting_tables(report=report,
+                                                            threshold=cve_threshold,
+                                                            tablefmt="simple")
+
+        print()
+        print(''.join(tables))
+
+        if send_notificaton:
+            # generate reporting table for notification
+            tables = whitesource.util.generate_reporting_tables(report=report,
+                                                                threshold=cve_threshold,
+                                                                tablefmt="html")
+
+            # get product risk report
+            ci.util.info('retrieving product risk report')
+            prr = client.get_product_risk_report(product_token=product_token).content
```

Comment:
simiar as to the other comment last time (mv to-json-conversion into client method), I would also in this case rather return the str, as opposed to exposing the http-request response, which:

a) makes consumption unncessarily cumbersome
b) exposes implementation details that a user of the client should not (have) to know or care about